spring-cloud-deployer-local/src/main/java/org/springframework/cloud/deployer/spi/local/LocalAppDeployer.java

```diff
@@ -137,6 +137,11 @@ public String deploy(AppDeploymentRequest request) {
 				Map<String, String> appInstanceEnv = new HashMap<>();
 				AppInstance instance = new AppInstance(deploymentId, i, appInstanceEnv, port);
 				ProcessBuilder builder = buildProcessBuilder(request, appInstanceEnv, args, Optional.of(i)).inheritIO();
+
+				if (request.getDeploymentProperties().containsKey(LocalDeployerProperties.DEBUG_PORT)) {
+					builder.command().add(1, this.buildRemoteDebugInstruction(request.getDeploymentProperties()));
```

Comment:
In the original PR from andy there was this tidbit 
```
command.add("-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address="+(debugPort+i));
```
The code is in a loop to deploy the apps based on the deployment property `count` so the port number needs to be difference for each instance that is deployed.